packages/server/src/services/Authentication/_utils.ts

```diff
@@ -2,12 +2,18 @@ import JWT from 'jsonwebtoken';
 import { ISystemUser } from '@/interfaces';
 import config from '@/config';
 
+interface IExtendedSystemUser extends ISystemUser {
+  oidc_access_token?: string;
+  oidc_id_token?: string;
+  oidc_refresh_token?: string;
+}
+
 /**
  * Generates JWT token for the given user.
- * @param {ISystemUser} user
+ * @param {IExtendedSystemUser} user
  * @return {string} token
  */
-export const generateToken = (user: ISystemUser): string => {
+export const generateToken = (user: IExtendedSystemUser): string => {
   const today = new Date();
   const exp = new Date(today);
   exp.setDate(today.getDate() + 60);
```

Comment:
@abouolia I have created new IExtendedSystemUser interface extending the current ISystemUser to store the oidc tokens which will be attached in request token so that it can be used to logout the oidc user and revoke the access tokens. Also it will be used in oidcSession middleware to check if user exist for that oidc access token.

Is it ok to store the oidc tokens in SystemUser table?
What will be the best way to work with the oidc tokens provided by the providers?